packages/backend/src/utils/npm.ts

```diff
@@ -0,0 +1,127 @@
+import { vscode } from "./vscodeProxy";
+import * as path from "path";
+
+export const isWin32 = platform() === "win32";
+const NPM = isWin32 ? "npm.cmd" : "npm";
+
+const NPM_REGISTRY_HOST = _.get(process, "env.NPM_CFG_REGISTRY", "http://registry.npmjs.com/");
+const SEARCH_QUERY_PREFIX = `${NPM_REGISTRY_HOST}-/v1/search?text=`;
+const SEARCH_QUERY_SUFFIX = "keywords:yeoman-generator &size=25&ranking=popularity";
+
+const CANCELED = "Action cancelled";
+const HAS_ACCESS = "Has Access";
+
+const enum choices {
+  set_defaul_location,
+  change_global_owner,
+  cancelled,
+}
+
+class Command {
+  private readonly globalNodeModulesPath;
+  private readonly globalPath;
+  private readonly CHANGE_OWNER_FOR_GLOBAL;
+  private readonly SET_DEFAULT_LOCATION;
+
+  constructor() {
+    this.globalNodeModulesPath = _.trim(execSync(`${NPM} root -g`).toString());
+
+    const nmLength = path.join(path.sep, "node_modules").length;
+    this.globalPath = this.globalNodeModulesPath.substring(0, this.globalNodeModulesPath.length - nmLength);
+
+    this.CHANGE_OWNER_FOR_GLOBAL = messages.change_owner_for_global(this.globalPath);
+    this.SET_DEFAULT_LOCATION = messages.set_default_location(customLocation.DEFAULT_LOCATION);
+  }
+
+  private getGenLocationParams(): string {
+    const customInstallationPath = customLocation.getPath();
+    return _.isEmpty(customInstallationPath) ? "-g" : `--prefix ${customInstallationPath}`;
+  }
+
+  public getGlobalNodeModulesPath(): string {
+    return this.globalNodeModulesPath;
+  }
+
+  public async execCommand(arg: string): Promise<any> {
+    return promisify(exec)(arg);
+  }
+
+  public getGensQueryURL(query: string, recommended: string): string {
+    return encodeURI(`${SEARCH_QUERY_PREFIX} ${query} ${recommended} ${SEARCH_QUERY_SUFFIX}`);
+  }
+
+  private async _install(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} install ${locationParams} ${genName}@latest`;
+    return this.execCommand(command);
+  }
+
+  public async uninstall(genName: string): Promise<any> {
+    const locationParams = this.getGenLocationParams();
+    const command = `${NPM} uninstall ${locationParams} ${genName}`;
+    return this.execCommand(command);
+  }
+
+  private async sudoExec(command: string) {
+    return new Promise((resolve, reject) => {
+      sudo.exec(command, {}, (err, script) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(script);
+        }
+      });
+    });
+  }
+
+  private async getAccessResult(): Promise<string | undefined> {
+    // we assume that if custom path set by an user is writable
+    if (_.isEmpty(customLocation.getPath())) {
+      const isWritable = await this.isPathWritable(this.globalPath);
+      if (!isWritable) {
+        return vscode.window.showInformationMessage(
+          messages.no_write_access(this.globalPath),
+          { modal: true },
+          this.CHANGE_OWNER_FOR_GLOBAL,
```

Comment:
This text does not appear for me when building locally!!
![Screenshot 2021-05-13 at 14 54 38](https://user-images.githubusercontent.com/10102054/118138209-f432cd80-b3fd-11eb-8e83-95bd6b7e2210.png)

Reply:
If I choose second option to install to custom location it fails to load the generator after install when triggering from the loadYeoman command. 
![Screenshot 2021-05-13 at 15 11 27](https://user-images.githubusercontent.com/10102054/118138308-14fb2300-b3fe-11eb-9426-a40fa70a7d6b.png)

gensMeta probably needs to be refreshed from the custom location.
meta

Reply:
It works on Windows.
Could you please debug and fid where exactly it fails ?

Reply:
see fix at for modal text https://github.com/SAP/yeoman-ui/pull/581#discussion_r632097491